catalog/app/components/Preview/loaders/Html.js

```diff
@@ -1,41 +1,49 @@
 import * as React from 'react'
 
 import cfg from 'constants/config'
-import * as AWS from 'utils/AWS'
 import AsyncResult from 'utils/AsyncResult'
 import { useIsInStack } from 'utils/BucketConfig'
 import { useStatusReportsBucket } from 'utils/StatusReportsBucket'
-import useMemoEq from 'utils/useMemoEq'
-
-import { PreviewData } from '../types'
+import useQuery from 'utils/useQuery'
 
+import * as IFrame from './IFrame'
 import * as Text from './Text'
 import FileType from './fileType'
 import * as utils from './utils'
 
+import BUCKET_CONFIG_QUERY from './IFrame/BrowsableBucketConfig.generated'
+
 export const detect = utils.extIn(['.htm', '.html'])
 
 export const FILE_TYPE = FileType.Html
 
-function IFrameLoader({ handle, children }) {
-  const sign = AWS.Signer.useS3Signer()
-  const src = useMemoEq([handle, sign], () =>
-    sign(handle, { ResponseContentType: 'text/html' }),
-  )
-  // TODO: issue a head request to ensure existence and get storage class
-  return children(
-    AsyncResult.Ok(PreviewData.IFrame({ src, modes: [FileType.Html, FileType.Text] })),
+// It's unsafe to render HTML in these conditions
+function useHtmlAsText(handle) {
+  const isInStack = useIsInStack()
+  const statusReportsBucket = useStatusReportsBucket()
+  return (
+    cfg.mode !== 'LOCAL' &&
+    !isInStack(handle.bucket) &&
+    handle.bucket !== statusReportsBucket
   )
 }
 
 export const Loader = function HtmlLoader({ handle, children }) {
-  const isInStack = useIsInStack()
-  const statusReportsBucket = useStatusReportsBucket()
-  return cfg.mode === 'LOCAL' ||
-    isInStack(handle.bucket) ||
-    handle.bucket === statusReportsBucket ? (
-    <IFrameLoader {...{ handle, children }} />
-  ) : (
-    <Text.Loader {...{ handle, children }} />
-  )
+  const renderHtmlAsText = useHtmlAsText(handle)
+  const bucketData = useQuery({
+    query: BUCKET_CONFIG_QUERY,
+    variables: { bucket: handle.bucket },
+    pause: renderHtmlAsText,
+  })
+  if (renderHtmlAsText) return <Text.Loader {...{ handle, children }} />
+  return bucketData.case({
+    fetching: () => children(AsyncResult.Pending()),
+    error: (e) => children(AsyncResult.Err(e)),
+    data: ({ bucketConfig: { browsable } }) =>
+      browsable ? (
+        <IFrame.BrowsableLoader {...{ handle, children }} />
+      ) : (
+        <IFrame.Loader {...{ handle, children }} />
```

Comment:
i'd call this something more specific, e.g. not `BrowsableLoader` vs `Loader`, but rather `LoaderBrowsable` vs `LoaderSigned` or smth like that